_posts/2022-01-21-dashboards-plugins-intro.markdown

```diff
@@ -0,0 +1,220 @@
+
+- Read the config file `opensearch_dashboards.yml`
+- Discover the plugins and construct a dependency tree. Both [core plugins](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/1.2/src/plugins) (plugins in the projects source that come with OpenSearch Dashboards) and external plugins
+- Load plugin specific config (if present)
+- Run the `osd:bootstrap` script within each plugin if its public code is not yet built.
+- Completes setup for each plugin loaded and starts the plugins
+
+The core system also manages each plugins lifecycle.
+
+## How to use them?
+
+Understanding the specifics of how the core system sets up and handles these plugins can get a little confusing. Thankfully working with plugins does not require an in depth understanding of its inner workings. With the help of some useful packages, the process of creating, loading, distributing plugins is made much simpler.
+
+### Creating a plugin
+
+To create a plugin, we can use the [`@osd/plugin-generator`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-plugin-generator/README.md) package to bootstrap a plugin template for us. To use it you can run in your terminal the command
+
+```sh
+node scripts/generate_plugin.js my_plugin_name # replace "my_plugin_name" with your desired plugin name
+```
+
+It will ask you a few questions about the plugin you want to generate and scaffold a plugin template that is ready to run in the appropriate plugin directory.
+
+### Loading a plugin
+
+To load a plugin into an instance of OpenSearch Dashboards, the plugin must be placed in one of the configured paths that the core system will search during its plugin discovery phase. There are 3 default paths from which plugins can be discovered:
+
+- `./src/plugins`
+- `./plugins`
+- `../opensearch-dashboards-extra`
+
+Placing the plugin in one of these paths will ensure that they are discovered by the core system. The default paths are defined by the [env](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/packages/osd-config/src/env.ts#L133-L138) class within the `@osd-config` package and discovered using the plugin service’s [discover method](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/server/plugins/discovery/plugins_discovery.ts#L65). Alternatively we can also define an additional config parameter `additionalPluginPaths` to define other explicit plugin paths. This is however not recommended for production use and will throw a warning when used outside of dev mode.
```

Comment:
- `Alternatively` -> `Alternatively,`
- What is "dev mode"? If that is hard to define reverse this and make it something like "...will throw a warning in production."